builder/builder-next/executor_linux.go

```diff
@@ -113,20 +113,20 @@ func (iface *lnInterface) init(c *libnetwork.Controller, n *libnetwork.Network)
 	defer close(iface.ready)
 	id := identity.NewID()
 
-	ep, err := n.CreateEndpoint(id, libnetwork.CreateOptionDisableResolution())
+	ep, err := n.CreateEndpoint(context.TODO(), id, libnetwork.CreateOptionDisableResolution())
```

Comment:
These are actual "TODO" (not "background"), correct?

could you put a `ctx := context.TODO()` at the start of this function instead if these should (when TODO is "done") share the same context ?

Reply:
I think it's good to do that for methods that don't have a `context.TODO()` somewhere in their body. But for those who do, I think that's going to make commits bigger and more painful to revert if it turns out things don't work nicely, so I won't do it everywhere.

Reply:
I started going through my commits one by one, I'm almost halfway through them and I start to think it's not a good idea.

In functions where there are already some `context.TODO()` / `context.Background()`, doing that in the same commits as the one where I add a new `context.TODO()` will make them harder to keep track of why things were done and harder to revert if ever needed.

In most of the functions where `context.TODO()` gets introduced for the first time, it looks like there are little to no chances to ever introduce another `context.TODO()`.

If I do that in separate commits, before introducing anything I'm afraid this PR will be pretty big, and once again, harder to review.